src/vs/workbench/api/common/extHost.api.impl.ts

```diff
@@ -1125,33 +1125,45 @@ export function createApiFactoryAndRegisterActors(accessor: ServicesAccessor): I
 			get fs() {
 				return extHostConsumerFileSystem.value;
 			},
-			registerFileSearchProvider: (scheme: string, provider: vscode.FileSearchProvider) => {
-				checkProposedApiEnabled(extension, 'fileSearchProvider');
-				return extHostSearch.registerFileSearchProviderOld(scheme, provider);
-			},
-			registerTextSearchProvider: (scheme: string, provider: vscode.TextSearchProvider) => {
-				checkProposedApiEnabled(extension, 'textSearchProvider');
-				return extHostSearch.registerTextSearchProviderOld(scheme, provider);
-			},
-			registerAITextSearchProvider: (scheme: string, provider: vscode.AITextSearchProvider) => {
-				// there are some dependencies on textSearchProvider, so we need to check for both
-				checkProposedApiEnabled(extension, 'aiTextSearchProvider');
-				checkProposedApiEnabled(extension, 'textSearchProvider');
-				return extHostSearch.registerAITextSearchProviderOld(scheme, provider);
-			},
-			registerFileSearchProviderNew: (scheme: string, provider: vscode.FileSearchProviderNew) => {
-				checkProposedApiEnabled(extension, 'fileSearchProviderNew');
-				return extHostSearch.registerFileSearchProvider(scheme, provider);
+			registerFileSearchProvider: (scheme: string, provider: vscode.FileSearchProvider | vscode.FileSearchProviderOld) => {
+				if (isProposedApiEnabled(extension, 'fileSearchProviderNew')) {
```

Comment:
@mjbvz is this the best way to test between using the old and new providers? Previously, I had different function calls to register both types, but I'm now just overloading the one call signature.

Reply:
Having separate methods would generally be cleaner but since there are existing users of the api I think this is fine